tests/test_file.py

```diff
@@ -414,6 +417,68 @@ def try_get():
                 url, requests.codes.bad_request)
         try_get()
 
+    @testmode.standalone
+    def test_file_get_checkout_exists(self):
+        """
+        Verifies checkout occurs on first get and not on second.
+        :return:
+        """
+        tempdir = tempfile.gettempdir()
+        self._test_file_get_checkout_exists(Replica.aws, "s3", self.s3_test_bucket,
+                                            S3Uploader(tempdir, self.s3_test_bucket))
+        self._test_file_get_checkout_exists(Replica.gcp, "gs", self.gs_test_bucket,
+                                            GSUploader(tempdir, self.gs_test_bucket))
+
+    def _test_file_get_checkout_exists(self, replica: Replica, scheme: str, test_bucket: str, uploader: Uploader):
+        src_key = generate_test_key()
+        src_data = os.urandom(1024)
+        with tempfile.NamedTemporaryFile(delete=True) as fh:
+            fh.write(src_data)
+            fh.flush()
+
+            uploader.checksum_and_upload_file(fh.name, src_key, "text/plain")
+
+        source_url = f"{scheme}://{test_bucket}/{src_key}"
+
+        file_uuid = str(uuid.uuid4())
+        bundle_uuid = str(uuid.uuid4())
+        version = datetime_to_version_format(datetime.datetime.utcnow())
+
+        # should be able to do this twice (i.e., same payload, different UUIDs)
+        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
+        url = str(UrlBuilder()
+                  .set(path="/v1/files/" + file_uuid)
+                  .add_query("replica", replica.name)
+                  .add_query("version", version))
+
+        now = datetime.datetime.now(pytz.utc)
+        # assert 301 redirect on first GET
+        self.assertGetResponse(url, requests.codes.moved, redirect_follow_retries=0)
+
+        @eventually(20, 2)
+        def test_checkout():
+            resp = self.assertGetResponse(
+                url, requests.codes.found, redirect_follow_retries=0)
+            html = BeautifulSoup(resp.body, 'html.parser')
```

Comment:
I think this can be done by inspecting the location header in the response.

Reply:
Got it. I didn't realize `assertGetResponse` returns a wrapper around the original response. Thanks!